### DISCUSSION TITLE
Selection Cuts (beginner)

### DISCUSSION THREAD

**Discussion by randomizer6542** [MadAnalysis/madanalysis5#193] [Q&A]
Hi,
So I am new to the software and was trying to replicate some studies for my practice and I came across a selection cuts that they implemented, I am unable to implement a few and I would like your help in how we can define those, I am using reco mode of ma5,

**The first cuts were:**
_pℓT > 25 GeV,     pj/bT > 30 GeV,    |ηi| < 2.5,    ∆Rij > 0.4 (i, j = ℓ, b, j)_
I implemented these by using
define i = l b
select PT(l) > 25
select PT(j) > 30
select PT(b) > 30
select ETA(i) < 2.5
select DELTAR(i,j) > 0.4

**and now the bottom selection cuts have me in difficulty:**
 _(Cut 1)_ There are three leptons in which at least two with positive charge and pT >
30 GeV, exactly one b-tagged jet with pT > 40 GeV and the event is rejected if the pT of
the subleading jet is greater than 25 GeV.

here I can easily set the lepton number but I don't know how to
_set the condition > in which at least two with positive charge
also does the subleading jet means j here? because it wouldn't make sense for it to be b again._

**(Cut 2)** The distance between the OSSF lepton pair should lie within ∆R(ℓ1, ℓ2) ∈
[0.4, 1.2] while the corresponding invariant mass is required to be |M(ℓ1ℓ2) − mZ| <
15 GeV.

For context, l1 = e+ e- , l2 = mu+ mu-; 
_I successfully implemented the first condition here but how can I implement |M(ℓ1ℓ2) − mZ| < 15 GeV ?_

also just a general question, if a wz background is disrupting your analysis, is it ok to lessen it's impact by directly applying cuts on its decays rather than the whole analysis?

Thank you for taking time to read,
Regards

**Comment by BFuks** [comment]
Hello,

First, beware of the difference between the following two cuts:
(1) reject PT(b) < 40
(2) reject (b) PT < 40
The former allows for the rejection of events in which b-jets with pt smaller than 40 GeV are found, where the latter takes the b-jet collection and remove from it any b-jet candidate with a transverse momentum smaller than 40 GeV. I have the impression that you are confused between both of them. A good way to understand what is actually done by MadAnalysis 5 is to explore the C++ code generated.

Now, to go back to your questions, complicated cuts can always be implemented directly in C++, through the so-called expert mode of the program. Often, that’s the only way to go if we target non-trivial possibilities. However, the first cut in which you are interested is not so complex. You could use:
  select N(l+)>=2 or N(l-)>=2
For the second cut, it is also easy as you could use
  select M(l+ l-) > 75 and M(l+ l-) < 105
or something similar.

The last of your questions consists of a physics question (and not a MadAnalysis 5 question) that cannot be answered without knowing your goals. I am sure that you will be able to figure out the answer by yourself.

Regards,

Benjamin

**Reply by randomizer6542** [reply]
Hi, Thanks for the answer

So, If i understand correctly 
reject PT(b) < 40   ->   This is for the main b-tagged jets while
reject (b) PT < 40   ->  This is for the sub-leading jets
correct?

Also thanks for all the other answers, it's quite clear to me and I will look into the expert mode as well because I think I will be needing it.

**Comment by BFuks** [comment]
Hi,

> So, If i understand correctly
> reject PT(b) < 40 -> This is for the main b-tagged jets while
> reject (b) PT < 40 -> This is for the sub-leading jets
> correct?
> 
Not really. The first cut applies to the whole event (selection or rejection) whereas the second one cleans a collection (i.e. it removes some b-jets from the b-jet collection).

Regards,

Benjamin

**Reply by randomizer6542** [reply]
hi,
So is there any command to control the sub-leading jets or is it can only be controlled by the expert mode?
Regards,

**Reply by jackaraz** [reply]
if you type `X[i]` to any object you can access each of the $p_T$ ordered element in your collection i.e. `j[1]` is the leading jet, `j[2]` is the second leading jet etc
